pkg/nodebootstrap/userdata_al2.go

```diff
@@ -54,6 +54,12 @@ func NewUserDataForAmazonLinux2(spec *api.ClusterConfig, ng *api.NodeGroup) (str
 
 	var scripts []string
 
+	if ng.SSH.EnableSSM != nil && *ng.SSH.EnableSSM {
```

Comment:
Are we sure this is run before the kubelet stuff?

Reply:
@michaelbeaumont I figured that as part of debugging failures that occur when bootstrapping having SSM installed first lets the users SSH on immediately, if for example the kubelet fails to start they won't be able to SSH on to debug

Reply:
Sorry I meant, is it clear whether or not the `addCommand` stuff ends up being run before the bootstrap script

Reply:
Oh sorry, I'm not sure. I saw that the `config.AddShellCommand` is used to add `PreBootstrapCommands`, so I assumed that meant it does run before the `bootstrap` script. Looking into it I do think adding it first does mean that the command runs before the bootstrap, as `addFilesAndScripts` just adds the file as an asset and appends the script as a command to the array of commands to run

Reply:
OK, I see `runcmd` runs only on first boot. I personally think it's cleaner to add the commands as a self-contained script vs calling `AddShellCommand`x3. As far as I can tell, we just need to call `RunScript` and make sure it has a name that comes earlier than `bootstrap.al2.sh` (although I don't like how `RunScript` calls both `AddScript` and `AddCommand`).

Reply:
Sure, I was torn between a new script vs adding commands so happy to refactor it over to a script 😄